server/src/test/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceShardsAllocatorTests.java

```diff
@@ -336,6 +341,143 @@ protected long currentNanoTime() {
         }
     }
 
+    public void testIndexCreationDuringLongDesiredComputation() {
```

Comment:
An IT doesn't seem to be straight forward (or feasible w/o larger refactoring). I went with this. If there is a simple way of emulating a long desired balance computation, let me know.